repo-scripts/size-analysis/bundle-analysis.ts

```diff
@@ -410,8 +427,64 @@ function createEntryFileContent(bundleDefinition: BundleDefinition): string {
   return contentArray.join('\n');
 }
 
+function createImportExport(
+  symbol: string,
+  modulePath: string,
+  symbolsCache: Set<string>
+): string[] {
+  const contentArray = [];
+
+  switch (symbol) {
+    case SpecialImport.Default: {
+      const nameToUse = createSymbolName('default_import', symbolsCache);
+      contentArray.push(`import ${nameToUse} from '${modulePath}';`);
+      contentArray.push(`console.log(${nameToUse})`); // prevent import from being tree shaken
+      break;
+    }
+    case SpecialImport.Namespace: {
+      const nameToUse = createSymbolName('namespace', symbolsCache);
+      contentArray.push(`import * as ${nameToUse} from '${modulePath}';`);
+      contentArray.push(`console.log(${nameToUse})`); // prevent import from being tree shaken
+      break;
+    }
+    case SpecialImport.Sizeeffect:
+      contentArray.push(`import '${modulePath}';`);
+      break;
+    default:
+      // named imports
+      const nameToUse = createSymbolName(symbol, symbolsCache);
+
+      if (nameToUse !== symbol) {
+        contentArray.push(
+          `export {${symbol} as ${nameToUse}} from '${modulePath}';`
+        );
+      } else {
+        contentArray.push(`export {${symbol}} from '${modulePath}';`);
+      }
+  }
+
+  return contentArray;
+}
+
+/**
+ * In case a symbol with the same name is already imported from another module, we need to give this symbol another name
+ * using "originalname as anothername" syntax, otherwise it returns the original symbol name.
+ */
+function createSymbolName(symbol: string, symbolsCache: Set<string>): string {
+  let nameToUse = symbol;
+  const max = 100;
+  while (symbolsCache.has(nameToUse)) {
+    nameToUse = `${symbol}_${Math.floor(Math.random() * max)}`;
```

Comment:
Super edge case but is it safer to increment instead of random?

Reply:
If there are more than 100 symbols with the same name, yes, but we likely will never encounter such scenario.
I chose to use random because it's faster than incrementing without maintaining additional cache. We could make incrementing O(1) by creating another cache that remembers the last number used for every symbol, but I think random is good enough.